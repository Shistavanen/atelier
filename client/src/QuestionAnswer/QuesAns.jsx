const React = require('react')
const axios = require('axios')
const QuestionView = require('./QuestionView.jsx');

class QuesAns extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      questionList: []
    }
  }

  getQuestionList() {
    axios.get('/questions')
      .then((res) => {
        const questionList = res.data;
        this.setState({questionList});
        //console.log('SUCESSFULLY GET BACK THE QUESTIONS LIST ', res);
      })
      .catch((err) => {
        console.log('FAIL TO GET BACK THE QUESTIONS LIST', err);
      })
  }

  componentDidMount(){
    this.getQuestionList();
  }

  render() {
    return(
      <div>
        <h1>Questions and Answers</h1>
        <input placeholder="HAVE A QUESTIONS? SEARCH FOR ANSWERS..."/>
        <div>
          {this.state.questionList.map(question =>
            <QuestionView question={question}/>
          )}

          <p></p>
        </div>
      </div>
    )
  }
}

module.exports = QuesAns;