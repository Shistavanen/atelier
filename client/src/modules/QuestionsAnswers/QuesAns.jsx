import React, { useState, useEffect } from 'react'
import ReactDOM from 'react-dom'
import axios from 'axios'
import QuestionView from './QuestionView.jsx'
import AddQuesForm from './AddQuesForm.jsx'

const QuesAns = ({product, productId}) => {
  const [questionList, setQuestionList] = useState([]);
  const [questionToShow, setQuestionToShow] = useState(2);
  const [showQuesForm, setShowQuesForm] = useState(false);
  const [searchTerm, setSearchTerm] = useState("");
  const [filtered, setFiltered] = useState([]);

  const handleCloseQuesForm = () => setShowQuesForm(false);
  const handleShowQuesForm = () => setShowQuesForm(true);

  const getQuestionList = () => {
   axios.get(`/questions/${productId}`)
      .then((res) => {
        console.log('SUCESSFULLY GET BACK THE QUESTIONS LIST ', res.data);
        setQuestionList([...res.data.results]);
      })
      .catch((err) => {
        console.log('FAIL TO GET BACK THE QUESTIONS LIST', err);
      })
  }

  const showMoreQuestion = () => {
    setQuestionToShow(questionList.length);
  }

  const handleChange = (e) => {
    //let filtered = [];
    setSearchTerm(e.target.value);
    console.log(" HERE IS WHAT USER SEARCH: ", searchTerm);
    if(searchTerm.length >= 2) {
      setFiltered(questionList.filter(question => question.question_body.includes(searchTerm)));
      console.log("here is the result list: ", filtered);
    }

    //console.log('FILTER', filtered);
  }
  const handleSearch = () => {
    setQuestionList(filtered);
  }

  useEffect(() => {
    getQuestionList();
  }, [])


  return(
    <div className="qaBackground">

      <h1>Questions and Answers</h1>
      <input value={searchTerm} placeholder="HAVE A QUESTIONS? SEARCH FOR ANSWERS..." onChange={handleChange} />
      <button onClick={handleSearch}>Search</button>

      <div>
        {questionList.slice(0, questionToShow).map(question =>
          <QuestionView key={question.question_id} question={question} product={product}/>
        )}
      <br />
      <br />
      {questionList.length > 2 ? <button onClick={() => {showMoreQuestion()}}>MORE ANSWERED QUESTIONS</button> : null}
      <button onClick={handleShowQuesForm}>ADD A QUESTION +</button>
      {showQuesForm && <AddQuesForm product={product} onHide={handleCloseQuesForm}/>}
      </div>

    </div>
  )
}

export default QuesAns;
//module.exports = QuesAns;
