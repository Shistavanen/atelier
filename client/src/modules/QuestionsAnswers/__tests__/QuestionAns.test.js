import React from 'react'
import renderer from 'react-test-renderer';
import QuesAns from '../QuesAns.jsx';

it('questions and answers matches snapshot', () => {
  const tree = renderer.create(<QuesAns />).toJSON();
  expect(tree).toMatchSnapshot();
});