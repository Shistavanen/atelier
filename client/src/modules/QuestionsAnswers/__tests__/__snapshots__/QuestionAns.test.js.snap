// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add question form matches snapshot 1`] = `
<div
  className="qaModalContainer"
>
  <button
    className="closeButton"
  >
    X
  </button>
  <div
    className="qaModalTitle"
  >
    <h3>
      Ask Your Question
    </h3>
    <h5>
      About the 
    </h5>
  </div>
  <div
    className="qaModalBody"
  >
    <label>
      Your Question *
    </label>
    <br />
    <textarea
      maxLength={1000}
      name="questionBody"
      onChange={[Function]}
      placeholder="Why did you like the product or not?"
      value=""
    />
    <p />
    <label>
      What is your nickname ? *
    </label>
    <br />
    <input
      maxLength={60}
      name="nickName"
      onChange={[Function]}
      placeholder="Example: jackson11!"
      value=""
    />
    <br />
    <span>
      For privacy reasons, do not use your full name or email address” will appear.
    </span>
    <p />
    <label>
      Your email *
    </label>
    <br />
    <input
      maxLength={60}
      name="email"
      onChange={[Function]}
      value=""
    />
    <br />
    <span>
      For authentication reasons, you will not be emailed.
    </span>
    <p />
    <br />
  </div>
  <div
    className="qaModalFooter"
  >
    <button
      className="submitButton"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
</div>
`;

exports[`question view matches snapshot 1`] = `
<div>
  <h3>
    Q: 
    Can I wash it?
  </h3>
  <span>
    Helpful? 
  </span>
  <a
    href="#"
    onClick={[Function]}
  >
    Yes
  </a>
  <span>
    (
    93
    )
  </span>
  <span>
     | 
  </span>
  <a
    href="#"
    onClick={[Function]}
  >
    Add Answer
  </a>
  <div>
    <h4>
      A: 
      I've thrown it in the wash and it seems fine
    </h4>
    <img
      src="https://images.unsplash.com/photo-1469504512102-900f29606341?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
    <img
      src="https://images.unsplash.com/photo-1510551310160-589462daf284?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1649&q=80"
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
    <div>
      by: 
      <span>
        skilover
      </span>
      , 
      <span>
        2017-01-04T00:00:00.000Z
      </span>
       |   Helpful?  
      <a
        href="#"
        onClick={[Function]}
      >
        Yes
      </a>
      <span>
        (
        22
        )
      </span>
      <span>
         | 
      </span>
      <a
        href="#"
        onClick={[Function]}
      >
        Report
      </a>
    </div>
  </div>
  <div>
    <h4>
      A: 
      It says not to
    </h4>
    <div>
      by: 
      <span>
        skilover
      </span>
      , 
      <span>
        2017-01-04T00:00:00.000Z
      </span>
       |   Helpful?  
      <a
        href="#"
        onClick={[Function]}
      >
        Yes
      </a>
      <span>
        (
        15
        )
      </span>
      <span>
         | 
      </span>
      <a
        href="#"
        onClick={[Function]}
      >
        Report
      </a>
    </div>
  </div>
  <br />
  <span
    onClick={[Function]}
  >
    Load More Answers
  </span>
  <br />
  *******************************************************************************
</div>
`;

exports[`question view matches snapshot 2`] = `
<div
  className="qaModalContainer"
>
  <button
    className="closeButton"
  >
    X
  </button>
  <div
    className="qaModalTitle"
  >
    <h3>
      Submit your Answer
    </h3>
    <h5>
      : 
      Can I wash it?
    </h5>
  </div>
  <div
    className="qaModalBody"
  >
    <label>
      Your Answer *
    </label>
    <br />
    <textarea
      maxLength={1000}
      name="ansBody"
      onChange={[Function]}
      value=""
    />
    <p />
    <br />
    <label>
      What is your nickname *
    </label>
    <br />
    <input
      maxLength={60}
      name="nickName"
      onChange={[Function]}
      placeholder="Example: jack543!"
      value=""
    />
    <br />
    <span>
      For privacy reasons, do not use your full name or email address” will appear.
    </span>
    <p />
    <br />
    <label>
      Your email *
    </label>
    <br />
    <input
      maxLength={60}
      name="email"
      onChange={[Function]}
      placeholder="Example: jack@email.com"
      value=""
    />
    <br />
    <span>
      For authentication reasons, you will not be emailed.
    </span>
    <p />
    <br />
    <label>
      Upload your photos 
    </label>
    <input
      multiple={true}
      onChange={[Function]}
      type="file"
    />
    <br />
  </div>
  <div
    className="qaModalFooter"
  >
    <button
      className="submitButton"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
</div>
`;

exports[`questions and answers matches snapshot 1`] = `
<div
  className="qaBackground"
>
  <h1>
    Questions and Answers
  </h1>
  <input
    onChange={[Function]}
    placeholder="HAVE A QUESTIONS? SEARCH FOR ANSWERS..."
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
  <div>
    <br />
    <br />
    <button
      onClick={[Function]}
    >
      ADD A QUESTION +
    </button>
  </div>
</div>
`;
